pos_lot_base/models/pos_config.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import fields, models
+
+
+class PosConfig(models.Model):
+
+    _inherit = "pos.config"
+
+    limited_lots_loading = fields.Boolean()
+    limited_lots_amount = fields.Integer(default=20000)
+    lot_load_background = fields.Boolean()
+
+    def get_limited_lots_loading(self, product_ids, lot_fields):
+        if not product_ids:
+            return []
+        query = """
+            WITH lm AS (
+                  SELECT lot_id,
+                         Max(write_date) date
+                    FROM stock_quant
+                    WHERE lot_id IS NOT NULL
+                GROUP BY lot_id
+            )
+               SELECT l.id
+                 FROM stock_production_lot l
+            LEFT JOIN product_product p ON l.product_id=p.id
+            LEFT JOIN product_template t ON p.product_tmpl_id=t.id
+            LEFT JOIN lm ON l.id=lm.lot_id
+                WHERE l.available_in_pos
+                    AND (l.company_id=%(company_id)s OR l.company_id IS NULL)
+                    AND p.id IN %(product_ids)s
+                    AND t.available_in_pos
```

Comment:
pos configuration has a warehouse that is used to take products, meaning not all lots will be available for sealing if they are in different warehouses